VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/SwitchDoesNotHandleAllEnumOptions/SwitchDoesNotHandleAllEnumOptionsAnalyzer.cs

```diff
@@ -26,24 +28,63 @@ internal static DiagnosticDescriptor Rule
 
         private void AnalyzeSymbol(SyntaxNodeAnalysisContext context)
         {
-            var switchBlock = (SwitchStatementSyntax) context.Node;
+            var switchBlock = (SwitchStatementSyntax)context.Node;
 
             var enumType = context.SemanticModel.GetTypeInfo(switchBlock.Expression).Type as INamedTypeSymbol;
             if (enumType == null || enumType.TypeKind != TypeKind.Enum)
             {
                 return;
             }
 
-            var caseLabels = switchBlock.Sections.SelectMany(l => l.Labels)
-                    .OfType<CaseSwitchLabelSyntax>()
-                    .Select(l => l.Value)
-                    .ToList();
+            var caseLabels = new List<ExpressionSyntax>();
 
-            var labelSymbols = caseLabels.Select(l => context.SemanticModel.GetSymbolInfo(l).Symbol);
+            foreach (var section in switchBlock.Sections)
+            {
+                foreach (var label in section.Labels)
+                {
+                    if (label.IsKind(SyntaxKind.CaseSwitchLabel))
+                    {
+                        caseLabels.Add(((CaseSwitchLabelSyntax)label).Value);
+                    }
+                }
+            }
+
+            var labelSymbols = new List<ISymbol>();
+            foreach (var label in caseLabels)
+            {
+                var symbol = context.SemanticModel.GetSymbolInfo(label).Symbol;
+                if (symbol == null)
+                {
+                    // potentially malformed case statement
+                    // or an integer being cast to an enum type
+                    return;
```

Comment:
I think you mean to use `continue` here, otherwise you're not going to enumerate the rest of your labels once an invalid one has been found. If that's intended (is it?) then it still doesn't work as it should because the rest of the method will go on with the existing labels that have already been added.